### PR TITLE
PR: TASKS IOPID-449, IOPID-452, IOPID-458, IOPID-477, IOPID-476

### PR DESCRIPTION
**Pull request by claudiodall**
**Feature:**

- New Header from MUI Italia 0.10.0
- Responsive Design (IOPID-476)
- Accordion Adjustment based on new design (IOPID-477)

**Flow Lock Profile:** 

- Pages (IOPID-452)
- Pages (IOPID-449)

**Flow Restore Profile:**

- Pages (IOPID-458)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,36 +1,48 @@
...
 
-Open [http://localhost:3000](http://localhost:3000) with your browser to see the result.
+## Technologies
+[![My Skills](https://skillicons.dev/icons?i=ts,react,next)](https://skillicons.dev)
+
+This is Next.js 13 with app route project
+## Prerequisites
+In order to run the ``io-web-profile`` front-end locally you need the following tool installed on your machine.
+
+- ``Node.js 18 ``
```
Comment (francescopersico) [inline README.md:18]: What you think about adding a .node-version file that can be used with nodenv?

File: README.md [hunk 2]
```diff
@@ -1,36 +1,48 @@
...
+    
+## Run
 
-## Deploy on Vercel
+To run the webapp locally:
 
-The easiest way to deploy your Next.js app is to use the [Vercel Platform](https://vercel.com/new?utm_medium=default-template&filter=next.js&utm_source=create-next-app&utm_campaign=create-next-app-readme) from the creators of Next.js.
+```bash
+# to run in dev mode
+yarn run dev
```
Comment (francescopersico) [inline README.md:41]: Have you checked that the local webapp is reachable after starting it in dev or production mode?

File: src/app/[locale]/(pages)/restoreCode/page.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+import { Button, Grid, Link, TextField } from '@mui/material';
+import { useRouter } from 'next/navigation';
+import { FAQ } from '../../_component/accordion/faqDefault';
+import { Introduction } from '../../_component/introduction/introduction';
+import { commonBackground } from '../../_utils/styles';
+
+const ReactivateCode = (): React.ReactElement => {
+  const { push } = useRouter();
+  const [restoreCode, setRestoreCode] = useState('');
+  const [notValidCode, setNotValidCode] = useState(false);
```
Comment (francescopersico) [inline src/app/[locale]/(pages)/restoreCode/page.tsx:14]:
```suggestion
  const [isCodeNotValid, setIsCodeNotValid] = useState(false);
```

File: src/app/[locale]/_component/accordion/faqDefault.tsx [hunk 4]
```diff
@@ -2,52 +2,107 @@ import React from 'react';
...
+interface FAQProps {
+  flow?: string;
+}
+
+interface FAQEntries {
+  summary: string;
+  details: string;
+}
+
+export function FAQ({ flow = 'DEFAULT' }: FAQProps) {
```
Comment (francescopersico) [inline src/app/[locale]/_component/accordion/faqDefault.tsx:14]: Why we are using a mix of normal and arrow function to define components? Pick a style and be consistent.

File: README.md (hunk 1 above)
Comment (OraldoDoci) [reply README.md]: Created the file .node-version.

File: README.md (hunk 2 above)
Comment (OraldoDoci) [reply README.md:41]: Readme updated.

File: src/app/[locale]/(pages)/profileBlock/page.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import { Button, Grid, Typography } from '@mui/material';
+import { FAQ } from '../../_component/accordion/faqDefault';
+import { BackButton } from '../../_component/backButton/backButton';
+import { IdpListOnApp } from '../../_component/idpListOnApp/idpListOnApp';
+import { Introduction } from '../../_component/introduction/introduction';
+import { commonBackgroundWithBack } from '../../_utils/styles';
+import { isIDPKnown } from '../../_utils/idps';
+
+const ProfileBlock = (): React.ReactElement => {
+  const renderSummary = (isIDPKnown: boolean) => {
```
[the remark this answers is not in the document]
Comment (OraldoDoci) [reply src/app/[locale]/(pages)/profileBlock/page.tsx]: extracted the function outside in a utils file.

File: src/app/[locale]/(pages)/restoreCode/page.tsx (hunk 3 above)
Comment (OraldoDoci) [reply src/app/[locale]/(pages)/restoreCode/page.tsx]: changed the use states names.

File: src/app/[locale]/(pages)/restoreCode/page.tsx [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+  const [errorMessage, setErrorMessage] = useState('');
+
+  // const reactivationCodePattern: string = '[0-9]{3}[ -]*[0-9]{3}[ -]*[0-9]{3}';
+  // const t = useTranslations('logout');
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
```
[the remark this answers is not in the document]
Comment (OraldoDoci) [reply src/app/[locale]/(pages)/restoreCode/page.tsx]: Changed the function name to handleRestoreCodeChange .

File: src/app/[locale]/_component/accordion/faqDefault.tsx (hunk 4 above)
Comment (OraldoDoci) [reply src/app/[locale]/_component/accordion/faqDefault.tsx]: Changed the component to an arrow Function.

**Comment by francescopersico** [comment]
Are you committing package-lock.json on purpose or is a mistake? We already have yarn.lock file.

File: README.md [hunk 7]
```diff
@@ -1,36 +1,55 @@
+
 # io-web-profile
 
-This is a [Next.js](https://nextjs.org/) project 
+`` 
+io-web-profile`` is the IO web platform that allows citizens to logout from the IO App session.
```
Comment (Ladirico) [inline README.md:5]: too much space before "io-web-profile"

File: README.md
Comment (Ladirico) [inline README.md:1]: miss the attachement to the documentation to start hub spid login (https://github.com/pagopa/hub-spid-login-ms/tree/master). This information is essential for a user who wants to star the entire environment locally

And need to explain the folder structure strategy

File: src/app/[locale]/(pages)/profileBlock/page.tsx
Comment (Ladirico) [inline src/app/[locale]/(pages)/profileBlock/page.tsx:1]: Miss margin left

File: src/app/[locale]/(pages)/profileBlock/utils.tsx [hunk 8]
```diff
@@ -0,0 +1,16 @@
+export const renderSummary = (isIDPKnown: boolean) => {
+  if (isIDPKnown === true) {
```
Comment (Ladirico) [inline src/app/[locale]/(pages)/profileBlock/utils.tsx:2]: in this case you can use if(isIDPKnown)

File: src/app/[locale]/(pages)/profileBlockSuccess/page.tsx
Comment (Ladirico) [inline src/app/[locale]/(pages)/profileBlockSuccess/page.tsx:1]: Miss margin at the left

File: src/app/[locale]/(pages)/profileRestore/page.tsx
Comment (Ladirico) [inline src/app/[locale]/(pages)/profileRestore/page.tsx:1]: Miss margin at the left

File: src/app/[locale]/(pages)/restoreThankYou/page.tsx
Comment (Ladirico) [inline src/app/[locale]/(pages)/restoreThankYou/page.tsx:1]: Miss margin at the left

**Review by Ladirico — CHANGES_REQUESTED** [review]

File: README.md (hunk 7 above)
Comment (OraldoDoci) [reply README.md]: removed the space.

**Pull request merged**